commons/src/main/java/org/infinispan/commons/dataconversion/Wrapper.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.commons.dataconversion;
+
+/**
+ *
+ * A Wrapper is used to decorate objects produced by the {@link Encoder} before storage.
```

Comment:
Some additional javadoc here would be helpful as well since this is an SPI. We also might want to mention why a user may use this class such as being able to apply a different equality/hashCode.